apps/webservice/src/app/[workspaceSlug]/_components/EditDeploymentDialog.tsx

```diff
@@ -24,41 +23,81 @@ import {
   FormField,
   FormItem,
   FormLabel,
+  FormMessage,
+  RootFormMessage,
 } from "@ctrlplane/ui/form";
 import { Input } from "@ctrlplane/ui/input";
 import { Textarea } from "@ctrlplane/ui/textarea";
 
 import { api } from "~/trpc/react";
 
-const deploymentForm = z.object({
-  id: z.string().uuid(),
-  name: z.string().min(3).max(255),
-  slug: z.string().min(3).max(255),
-  description: z.string().optional(),
+const deploymentFormSchema = z.object({
+  systemId: z.string().uuid({ message: "Invalid system ID format." }),
+  id: z.string().uuid({ message: "Invalid ID format." }),
+  name: z
+    .string()
+    .min(3, { message: "Name must be at least 3 characters long." })
+    .max(255, { message: "Name must be at most 255 characters long." }),
+  slug: z
+    .string()
+    .min(3, { message: "Slug must be at least 3 characters long." })
+    .max(255, { message: "Slug must be at most 255 characters long." }),
+  description: z
+    .string()
+    .max(255, { message: "Description must be at most 255 characters long." })
+    .optional()
+    .refine((val) => !val || val.length >= 3, {
+      message: "Description must be at least 3 characters long if provided.",
+    }),
 });
 
-type DeploymentFormValues = z.infer<typeof deploymentForm>;
+type DeploymentFormValues = z.infer<typeof deploymentFormSchema>;
 
 export const EditDeploymentDialog: React.FC<
   DeploymentFormValues & { children?: React.ReactNode }
-> = ({ id, name, slug, description, children }) => {
-  const router = useRouter();
-  const update = api.deployment.update.useMutation();
+> = ({ systemId, id, name, slug, description, children }) => {
   const [open, setOpen] = useState(false);
+  const utils = api.useUtils();
 
   const form = useForm({
-    resolver: zodResolver(deploymentForm),
-    defaultValues: { id, name, slug, description },
+    resolver: zodResolver(deploymentFormSchema),
+    defaultValues: { systemId, id, name, slug, description },
+    mode: "onChange",
   });
 
-  const onSubmit = form.handleSubmit(async (data) => {
-    setOpen(false);
-    const isDataChanged = !isEqual(data, { name, slug, description });
-    if (!isDataChanged) return;
+  const update = api.deployment.update.useMutation({
+    onError: (error) => {
+      if (error.message.includes("violates unique constraint")) {
+        form.setError("root", {
+          type: "manual",
+          message:
+            "A deployment with this slug already exists. Please choose a different slug.",
+        });
+        return;
+      }
+      form.setError("root", {
+        type: "manual",
+        message: "An unexpected error occurred. Please try again.",
+      });
+    },
+    onSuccess: () => {
+      utils.deployment.invalidate();
+      utils.deployment.bySystemId.invalidate(systemId);
```

Comment:
@adityachoudhari26, can you give me some pointers?

I tried async and not as we are using it in different places.

The initial call that generates the table where the name needs to be updated is in 
apps/webservice/src/app/[workspaceSlug]/systems/[systemSlug]/deployments/page.tsx 
I beleive. 

Thx